### PR TITLE
AN-27024

### PR DESCRIPTION
**Pull request by davidhanrhize**
Added config documentation for Audit, Agent, BPMN, and Core microservices.

### CODE REVIEW

File: content/reference/agent-configuration.md [hunk 1]
```diff
@@ -22,6 +22,101 @@ As the communication bridge between the Rhize Data Hub and your plant, the agent
...
+| Username            | Authenticates through a `username` and `password` in the config file, or through a Kubernetes secret.                                                                                                                           |
 | Certificate         | Uses the certificate on disk specified in the  `OPCUA.CertFile` and `OPCUA.KeyFile` configs. If no certificate exists and the config specifies the `OPCUA.GenCert` property as `true`, automatically generates one. |
+
+## logging
+
+ Logs the configurations to the console.
+
+| Attributes | Description |   
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `type`                | Specifies the logging configuration type: `json`, `multi`, or console by default. <br />(Default: `console`) |
```
Comment (MattDodsonEnglish) [inline content/reference/agent-configuration.md:35]:
```suggestion
| `type`                | Specifies the logging configuration type: `json`, `multi`, or `console`. <br />(Default: `console`) |
```

File: content/reference/agent-configuration.md [hunk 2]
```diff
@@ -22,6 +22,101 @@ As the communication bridge between the Rhize Data Hub and your plant, the agent
...
+## OIDC
+
+ Configurations for Keycloak authentication and connection with OpenID Connect.
+
+| Attribute | Description |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `serverUrl`         | The URL of the OpenID Connect server. <br />(Default: `http://localhost:8090`) |
+| `realm`         | Identifies the authentication domain for which the authentication request is being made. |
+| `client_id`         | The unique identifier assigned to the client application by the OIDC server. |
+| `client_secret`         | Used to authenticate the client alongisde the client ID when making confidential requests. |
```
Comment (MattDodsonEnglish) [inline content/reference/agent-configuration.md:59]:
```suggestion
| `client_secret`         | Used to authenticate the client alongside the client ID when making confidential requests. |
```

File: content/reference/agent-configuration.md [hunk 3]
```diff
@@ -22,6 +22,101 @@ As the communication bridge between the Rhize Data Hub and your plant, the agent
...
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `serverUrl`         | The URL of the OpenTelemetry server.  <br />(Default: `localhost:4317`) |   
+
+## OPCUA 
+
+| Attribute | Description |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `DiscoveryUrl`         | The URL to locate and connect to OPC UA servers on a network. <br />(Default: `opc.tcp://localhost:4840`) |
+| `Endpoint`         | The URL of the OPC UA service server.  <br />(Default: `opc.tcp://localhost:4840`) |
+| `Username`         | The username credentials to authenticate with the OP CUA server. |
```
Comment (MattDodsonEnglish) [inline content/reference/agent-configuration.md:75]:
```suggestion
| `Username`         | The username credentials to authenticate with the OPC UA server. |
```

File: content/reference/agent-configuration.md [hunk 4]
```diff
@@ -22,6 +22,101 @@ As the communication bridge between the Rhize Data Hub and your plant, the agent
...
+| `Password`         | The password credentials to authenticate with the. |
```
Comment (MattDodsonEnglish) [inline content/reference/agent-configuration.md:76]:
```suggestion
| `Password`         | The password credentials to authenticate with the OPC UA server. |
```

File: content/reference/agent-configuration.md [hunk 5]
```diff
@@ -22,6 +22,101 @@ As the communication bridge between the Rhize Data Hub and your plant, the agent
...
+| Attribute | Description |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `ProtocolQueueType`         | The type of queue used for buffering communication protocol data.  <br />(Default: `0`) |   
+
+## HEALTH
+
+| Attribute | Description |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `PollInterval`         | The frequency of scans for component status and health.  <br />(Default: `1000`) |   
+| `SubscriptionTimeout`         | The maximum duration for waiting to receive updates from subscribed data sources.  <br />(Default: `60000`) |
```
Comment (MattDodsonEnglish) [inline content/reference/agent-configuration.md:93]:
```suggestion
| `SubscriptionTimeout`         | The maximum duration to wait to receive updates from subscribed data sources.  <br />(Default: `60000`) |   
```

File: content/reference/agent-configuration.md [hunk 6]
```diff
@@ -22,6 +22,101 @@ As the communication bridge between the Rhize Data Hub and your plant, the agent
...
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `PollInterval`         | The frequency of scans for component status and health.  <br />(Default: `1000`) |   
+| `SubscriptionTimeout`         | The maximum duration for waiting to receive updates from subscribed data sources.  <br />(Default: `60000`) |   
+| `SubscriptionMaxCount`         | The maximum number of concurrent subscriptions for monitoring.  <br />(Default: `5`) |   
+
+## MQTT
+
+| Attribute | Description |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `Version`         | The version of MQTT used: `5.0` or `3.1.1`. <br />(Default: `3.1.1`) |
```
Comment (MattDodsonEnglish) [inline content/reference/agent-configuration.md:100]: @Jarrah-libremfg Does the agent accept 5.0 yet?

File: content/reference/agent-configuration.md [hunk 7]
```diff
@@ -22,6 +22,101 @@ As the communication bridge between the Rhize Data Hub and your plant, the agent
...
+| Attribute | Description |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `Version`         | The version of MQTT used: `5.0` or `3.1.1`. <br />(Default: `3.1.1`) |   
+| `ClientId`         | The ID used in the MQTT broker.  <br />(Default: `mqtt-client`) |   
+| `Endpoint`         | The URL of the MQTT broker.  <br />(Default: `mqtt://localhost:1883`) |   
+| `Username`         | The username credentials to authenticate with the MQTT broker. |   
+| `Password`         | The password credentials to authenticate with the MQTT broker. |   
+| `DecomposeJSON`         | Enables or disables JSON payload decomposition into individual data fields.  <br />(Default: `false`) |   
+| `TimestampField`         | The field to search to return timestamp information.  <br />(Default: `timestamp`) |   
+| `RequestTimeout`         | The maximum duration for waiting to receive a response to an MQTT request from the broker.  <br />(Default: `10`) |
```
Comment (MattDodsonEnglish) [inline content/reference/agent-configuration.md:107]:
```suggestion
| `RequestTimeout`         | The maximum duration to wait to receive a response to an MQTT request from the broker.  <br />(Default: `10`) |   
```

File: content/reference/audit-configuration.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+## influxdb
+
+ A time-series database that is used in conjunction with Grafana designed for handling time-stamped data, such as metrics, events, and logs, that change over time.
+
+| Attributes          | Description                                                                                                                                                                                    |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `serverUrl`         | The URL of the InfluxDB server. <br />(Default: `http://localhost:8086`)                                                                                                                       |    
+| `token`             | The authentication token used to authenticate requests to the InfluxDB server. <br />(Default: `my-token`)                                                                                     |
```
Comment (MattDodsonEnglish) [inline content/reference/audit-configuration.md:55]:
```suggestion
| `token`             | The authentication token to authenticate requests to the InfluxDB server. <br />(Default: `my-token`)                                                                                     |               
```

File: content/reference/audit-configuration.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+| `token`             | The authentication token used to authenticate requests to the InfluxDB server. <br />(Default: `my-token`)                                                                                     |               
+
+## pg
+
+ PostgreSQL is a general-purpose relational database management system that supports a wide range of features and data types.
+
+| Attributes          | Description                                                                                                                                                                                    |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `host`              | The host name of the PostgreSQL database server to which the client application will connect. <br />(Default: `dbname`)                                                                        |
```
Comment (MattDodsonEnglish) [inline content/reference/audit-configuration.md:63]:
```suggestion
| `host`              | The host name of the PostgreSQL database server to which the client application connects. <br />(Default: `dbname`)                                                                        |    
```

File: content/reference/audit-configuration.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+## pg
+
+ PostgreSQL is a general-purpose relational database management system that supports a wide range of features and data types.
+
+| Attributes          | Description                                                                                                                                                                                    |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `host`              | The host name of the PostgreSQL database server to which the client application will connect. <br />(Default: `dbname`)                                                                        |    
+| `user`              | The username used to authenticate with the PostgreSQL database server.                                                                                             |
```
Comment (MattDodsonEnglish) [inline content/reference/audit-configuration.md:64]:
```suggestion
| `user`              | The username to authenticate with the PostgreSQL database server.                                                                                             |         
```

File: content/reference/bpmn-configuration.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+categories: ["reference"]
+description: Authentication types for the Rhize BPMN
+weight: 900
+menu:
+  main:
+    parent: reference
+    identifier:
+---
+
+The Rhize BPMN acts as the tailored engine for processing low-code workflows designed within the BPMN UI. The configurations manage the connection and data flow on the BPMN engine to the other Rhize microservices.
```
Comment (MattDodsonEnglish) [inline content/reference/bpmn-configuration.md:12]:
```suggestion
The Rhize BPMN acts as the tailored engine for processing low-code workflows designed within the [BPMN UI]({{< relref "/how-to/bpmn" >}}. The configurations manage the connection and data flow on the BPMN engine to the other Rhize microservices. 
```

File: content/reference/bpmn-configuration.md [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+## http
+
+ All HTTP configurations are measured in `seconds`.
+
+| Attribute | Description |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `ReadHeaderTimeout`   | Wait duration for the request header to be fully read before timing out. <br />(Default: `10`)           |
+| `ReadTimeout`         | Wait duration for the entire request to be read before timing out. <br />(Default: `15`)                 |
+| `WriteTimeout`        | Wait duration for the entire response to be written before timing out. <br />(Default: `10`)             |
+| `IdleTimeout`         | Wait duration for the next request whil the connection is idle before timing out. <br />(Default: `30`)  |
```
Comment (MattDodsonEnglish) [inline content/reference/bpmn-configuration.md:23]:
```suggestion
| `IdleTimeout`         | Wait duration for the next request while the connection is idle before timing out. <br />(Default: `30`)  |
```

File: content/reference/bpmn-configuration.md [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+## OIDC
+
+ Configurations for Keycloak authentication and connection with OpenID Connect.
+
+| Attribute | Description |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `serverUrl`         | The URL of the OpenID Connect server. <br />(Default: `http://localhost:8090`) |
+| `realm`         | Identifies the authentication domain for which the authentication request is being made. |
+| `client_id`         | The unique identifier assigned to the client application by the OIDC server. |
+| `client_secret`         | Used to authenticate the client alongisde the client ID when making confidential requests. |
```
Comment (MattDodsonEnglish) [inline content/reference/bpmn-configuration.md:84]:
```suggestion
| `client_secret`         | Used to authenticate the client alongside the client ID when making confidential requests. |
```

File: content/reference/core-configuration.md [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+categories: ["reference"]
+description: Configuration for the Rhize core
+weight: 900
+menu:
+  main:
+    parent: reference
+    identifier:
+---
+
+ Core acts as a bespoke edge agent, overseeing data sources such as OPC-UA servers, and managing the publication and subscription of topics within the NATS messaging system.
```
Comment (MattDodsonEnglish) [inline content/reference/core-configuration.md:12]:
```suggestion
The Core service oversees data sources such as OPC-UA servers and manages the publication and subscription of topics within the NATS messaging system.
```

Comment (MattDodsonEnglish) [reply content/reference/core-configuration.md:12]: Simplifiying to remove any confusion with the Agent service, and most of the Core behavior is really interesting only internally, so no need to explain too much.

File: content/reference/core-configuration.md [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+
+## logging
+
+ Logs the configuration to the console.
+
+| Attributes          | Description                                                                                                                                                                                    |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `type`              | Specifies the logging configuration type: `json`, `multi`, or console by default. <br />(Default: `console`)                                                                                   |
```
Comment (MattDodsonEnglish) [inline content/reference/core-configuration.md:20]:
```suggestion
| `type`              | Specifies the logging configuration type: `json`, `multi`, or `console`. <br />(Default: `console`)                                                                                   |                                                                                                                                                 
```

File: content/reference/core-configuration.md [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `serverUrl`         | The URL of the OpenTelemetry server. <br />(Default: `localhost:4317`)                                                                                                                         | 
+| `samplingRate`      | The sampling rate for for traces. <br />(Default: `1`)                                                                                                                                         | 
+
+## SECRET
+
+| Attributes          | Description                                                                                                                                                                                    |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `KEY`               | The SECRET KEY used for authorization within Core.       |
```
Comment (MattDodsonEnglish) [inline content/reference/core-configuration.md:57]: Why is secret key in caps?

**Review by MattDodsonEnglish — CHANGES_REQUESTED** [review]
This is excellent! Well done @davidhanrhize @mattprincevrhize .

All of my comments are just nits to get the character length down. But I found very little to chop :+1: 

One other small nit, please put the headings in monospace. (backtics). This will emphasize that it is code configuration. Some of these names could look a little jarring or incorrect out of context.

I think this script should do it:


```bash
sed -i "s/## \([^ ]*\)$/## \`\\1\`/g" content/reference/*-configuration.md 
```

After that and my nits, let's merge! 

My other comment is that I think we could merge all these in one category called "Microservice configuration". But don't worry, I can do that after.

File: content/reference/agent-configuration.md (hunk 6 above)
Comment (davidhanrhize) [reply content/reference/agent-configuration.md:100]: From what we're aware of, there is a MqttAdapterV5.go file implementation in progress, and the code reads both 3.1.1 and 5.0 versions, but the 5 case just raises an unsupported message currently.

File: content/reference/core-configuration.md (hunk 16 above)
Comment (davidhanrhize) [reply content/reference/core-configuration.md:57]: The secret key config in Core is written like that in the config file, so we wanted to replicated its syntax exactly as seen.

**Pull request merged**